lib/jwt.js

```diff
@@ -0,0 +1,173 @@
+var urlDecodeB64 = function(data) {
+  return new Buffer(data, 'base64').toString('ascii');
+}
+
+var isNumber = function(n) { return typeof n === 'number' };
+
+/**
+ * Decodes a string token into the 3 parts, throws if the format is invalid
+ * @param token
+ */
+var decode = function(token) {
+  var parts = token.split('.');
+
+  if (parts.length !== 3) {
+    throw new Error('ID token could not be decoded')
+  }
+
+  var header = parts[0];
+  var payload = parts[1];
+  var signature = parts[2];
+
+  return {
+    _raw: token,
+    header: JSON.parse(urlDecodeB64(header)),
+    payload: JSON.parse(urlDecodeB64(payload)),
+    signature: signature
+  };
+}
+
+/**
+ * Verifies an idToken is valid and matches the options passed in
+ * @param idToken
+ * @param options
+ */
+var verify = function(idToken, options) {
+  if (!idToken) {
+    throw new Error('ID token is required but missing');
+  }
+
+  var decoded = decode(idToken);
+
+  if (!decoded.payload.iss) {
+    throw new Error(
+      'Issuer (iss) payload must be a string present in the ID token'
+    );
+  }
+
+  if (decoded.payload.iss !== options.iss) {
+    throw new Error(
+      `Issuer (iss) payload mismatch in the ID token; expected "${options.iss}", found "${decoded.payload.iss}"`
+    );
+  }
+
+  if (!decoded.payload.sub) {
+    throw new Error(
+      'Subject (sub) payload must be a string present in the ID token'
+    );
+  }
+
+  if (decoded.header.alg !== 'RS256' && decoded.header.alg !== 'HS256') {
+    throw new Error(
+      `Signature algorithm of "${decoded.header.alg}" is not supported. Expected the ID token to be signed with "RS256" or "HS256".`
+    );
+  }
+
+  if (
```

Comment:
This is all on a single line in the sample ... any reason to have it broken out like this?